src/components/history/GitOpsDeploymentHistory.tsx

```diff
@@ -0,0 +1,124 @@
+import './GitOpsDeploymentHistory.scss';
+
+import * as _ from 'lodash';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { RouteComponentProps } from 'react-router-dom';
+
+import {
+  ListPageFilter,
+  RowFilter,
+  useListPageFilter,
+  VirtualizedTable,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { LoadingBox } from '@patternfly/quickstarts';
+
+import GitOpsEmptyState from '../GitOpsEmptyState';
+import { GitOpsHistoryData } from '../utils/gitops-types';
+import { getApplicationsBaseURI, getEnvData, getPipelinesBaseURI } from '../utils/gitops-utils';
+import useDefaultSecret from '../utils/useDefaultSecret';
+import useEnvDetails from '../utils/useEnvDetails';
+import { GitOpsDeploymentHistoryColumns } from './GitOpsDeploymentHistoryColumns';
+import { GitOpsDeploymentHistoryTableRow } from './GitOpsDeploymentHistoryTableRow';
+
+type GitOpsDeploymentHistoryProps = RouteComponentProps<{ appName?: string }>;
+const GitOpsDeploymentHistory: React.FC<GitOpsDeploymentHistoryProps> = ({ match }) => {
+  const { appName } = match.params;
+  const [secretNS, secretName] = useDefaultSecret();
+  const pipelinesBaseURI = getPipelinesBaseURI(secretNS, secretName);
+  const searchParams = new URLSearchParams(location.search);
+  const manifestURL = searchParams.get('url');
+  const applicationBaseURI = getApplicationsBaseURI(appName, secretNS, secretName, manifestURL);
+  const [envs, emptyStateMsg] = useEnvDetails(appName, manifestURL, pipelinesBaseURI);
```

Comment:
Similarly, the API will be called twice.  See comment in GitOpsDetails.

Reply:
Yes, I called it 3 times which is not a good practice. Do you have any suggestion on this?

Reply:
Resolved as in GitOpsDetails